api/controllers/subscriber.js

```diff
@@ -108,6 +127,60 @@ async function getSubscriber(req, res) {
       }
     }
 
+    if (subscriber.transaction?.platform === 'paypal' &&
+      subscriber.transaction?.subscriptionId) {
+      let status = '';
+      let expiryDate = '';
+      let remoteData = '';
+      let nativePurchase = '';
+      try {
+        // get subscription from paypal API
+        remoteData = await paypal.getSubscriptionDetails(subscriber.transaction.subscriptionId);
+        status = remoteData.status;
+        if (status.toLowerCase() === 'cancelled') status = 'canceled';
+        expiryDate = remoteData.billing_info?.next_billing_time;
+        nativePurchase = remoteData;
+      } catch (err) {
+        console.log(err.message);
+      }
+      if (status) {
+        subscriber.status = status;
+        subscriber.transaction.expiryDate = expiryDate;
+        subscriber.transaction.nativePurchase = nativePurchase;
+        if (remoteData)
+          subscriber.save(function (err, subscr) {
```

Comment:
Needs to await the save function. If not can return the 200 state on line 184 and fails after save. This breaks the server. Same on line 97. (Both lines on PR file)

Reply:
fixed